feat: Initial troubleshooting page for RN (#3843)
getsentry/sentry-docs: src/platforms/react-native/troubleshooting.mdx

---
title: Troubleshooting
description: "Troubleshoot and resolve common issues with the React Native SDK."
sidebar_order: 1000
---

## Source Maps

> When there is an issue with source maps on your event, there will usually be an error dialog at the top of the event page on your dashboard:

Reviewer: ```suggestion
When there is an issue with source maps on your event, there will usually be an error dialog at the top of the Issue Details page in [sentry.io](https://sentry.io):
```

> There was 1 problem processing this event: Source code was not found

> Common fixes to this are:

Reviewer: ```suggestion
Common fixes to this issue include:
```

> - Make sure that both the `release` and `dist` and are set on the event by [setting it in the call to `init`](/platforms/react-native/configuration/releases/#bind-the-version).

Reviewer: ```suggestion
- Checking that both the `release` and `dist` and are set on the event by [setting it in the call to `init`](/platforms/react-native/configuration/releases/#bind-the-version). 
```

> - Make sure that the source maps are correctly uploaded with the exact same `release` and `dist` values as the event.

Reviewer: ```suggestion
- Checking that source maps are correctly uploaded with the exact same `release` and `dist` values as the event. They **must match** for source maps to upload correctly.
```

> - If you set a custom `release` and `dist` make sure that you manually upload the sourcemaps as our automatic sourcemap upload script will not detect the custom values.

Reviewer: ```suggestion
- If you set a custom `release` and `dist`, confirming that you manually uploaded the sourcemaps since our automatic source map upload script will not detect the custom values.
```

> - Make sure that you are not disabling the `RewriteFrames` default integration by filtering it out or setting `defaultIntegrations: false`.

Reviewer: ```suggestion
- Checking that you are not disabling the `RewriteFrames` default integration by filtering it out or setting `defaultIntegrations: false`.
```

Reviewer: We don't seem to have `RewriteFrames` (or any integrations) documented for RN. Do we want to link to the content in JS?

Author: Yes we should!

Author: @PeloWriter Actually, additional thought is that we should include the integrations content in RN as well. All the pluggable integrations and most (but not all) of the default integrations are in RN as well.

Reviewer: That makes sense. The sad part is that we don't have the ability (yet) to inherit from siblings (that is, since RN is at the same level as the JS content, we can't just inherit the JS content). The best way (it's a kludge for sure) to resolve this is to build those pages out, then `include` them in both RN and JS. Alternately, just linking works for now until we resolve this issue.

[For more details, please read the guide on React Native source maps.](/platforms/react-native/sourcemaps)

## iOS Build Script Failed

> Try passing `--force_foreground` to the Sentry CLI command in the build script. This is possibly a bug with our CLI that we are investigating.

Reviewer: Can we add a link to the the issue that tracks this potential bug? That way, as soon as it's resolved, if we don't immediately update this content, users will know of the resolution.

```bash
# ...
../node_modules/@sentry/cli/bin/sentry-cli react-native xcode \
  ../node_modules/react-native/scripts/react-native-xcode.sh --force_foreground
```

## Release Health

> - If your release health statistics are being attributed to the wrong release, make sure you [pass the `release` and `dist` to the `init` call](/platforms/react-native/configuration/releases/#bind-the-version).

Reviewer: ```suggestion
- If your release health statistics are being attributed to the wrong release, confirm that you [pass the `release` and `dist` to the `init` call](/platforms/react-native/configuration/releases/#bind-the-version).
```
